lib/internal/modules/esm/assert.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+const {
+  ArrayPrototypeFilter,
+  ArrayPrototypeIncludes,
+  ObjectCreate,
+  ObjectValues,
+  ObjectPrototypeHasOwnProperty,
+} = primordials;
+const { validateString } = require('internal/validators');
+
+const {
+  ERR_IMPORT_ASSERTION_TYPE_FAILED,
+  ERR_IMPORT_ASSERTION_TYPE_MISSING,
+  ERR_IMPORT_ASSERTION_TYPE_UNSUPPORTED,
+} = require('internal/errors').codes;
+
+// The HTML spec has an implied default type of `'javascript'`.
+const kImplicitAssertType = 'javascript';
+
+/**
+ * Define a map of module formats to import assertion types (the value of
+ * `type` in `assert { type: 'json' }`).
+ * @type {Map<string, string>}
+ */
+const formatTypeMap = {
+  '__proto__': null,
```

Comment:
should the quotes be here? keys are usually quoted only when necessary, in general, but dunder proto i think means something different when quoted.
```suggestion
  __proto__: null,
```

Reply:
All the keys in this object have unnecessary quotes, however they're in the origin commit, so if you want to open a PR to remove them, I can add the commit to this PR once it's updated on `master`.

Reply:
I did a quick check in the REPL. `__proto__: null` and `'__proto__': null` seem equivalent:

```
> a = { __proto__: { value: 42 } }
{}
> a.value
42
> a = { '__proto__': { value: 42 } }
{}
> a.value
42
```

Reply:
Maybe it’s only when computed that it’s different. Thanks for checking!